### PR TITLE
fix(knative): Integration in error state when knative SinkBinding mode=inclusion

### PR DESCRIPTION
**Pull request by claudio4j**
https://github.com/apache/camel-k/issues/4119

**Release Note**
```release-note
knative trait sets `bindings.knative.dev/include=true` label to the namespace when there is a SinkBinding.
```

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: e2e/yaks/common/knative-sinkbinding-http/sinkbinding.feature [hunk 1]
```diff
@@ -0,0 +1,9 @@
+Feature: Camel K can run source HTTP endpoint in sinkbinding mode
+
+  Background:
+    Given Kubernetes resource polling configuration
+      | maxAttempts          | 1   |
+      | delayBetweenAttempts | 500 |
+
+  Scenario: Integration knative-service starts with no errors
+    Given wait for condition=Ready on Kubernetes custom resource integration/rest2channel in integration.camel.apache.org/v1
```
Comment (christophd) [inline e2e/yaks/common/knative-sinkbinding-http/sinkbinding.feature:9]: Why not use the Camel K step `Given Camel K integration rest2channel is running`?

**Review by christophd — COMMENTED** [review]
Looks good @claudio4j many thanks! I remember that we have already fixed something regarding namespace labelling for Knative sink binding injection in the past. Do you know why this came back as a regression?

**Review by squakez — APPROVED** [review]

File: config/rbac/operator-role.yaml [hunk 2]
```diff
@@ -71,6 +71,7 @@ rules:
   - configmaps
   - secrets
   - serviceaccounts
+  - namespaces
```
Comment (astefanutti) [inline config/rbac/operator-role.yaml:74]: It doesn't seem like a best practice to grant all verbs on `namespaces` resources, especially `delete` and `deletecollection`.  Could this be refined to the minimal set of verbs?

**Review by astefanutti — CHANGES_REQUESTED** [review]

**Comment by claudio4j** [comment]
> I remember that we have already fixed something regarding namespace labelling for Knative sink binding injection in the past. Do you know why this came back as a regression?

It was fixed only on 1.8 downstream at the time.

File: config/rbac/operator-role.yaml [hunk 3]
```diff
@@ -80,6 +80,13 @@ rules:
   - patch
   - update
   - watch
+- apiGroups:
```
Comment (astefanutti) [inline config/rbac/operator-role.yaml:83]: What about adding it in the `camel-k-operator-knative` role, as it's strictly needed for Knative, to keep the RBAC modularity consistent?

Comment (claudio4j) [reply config/rbac/operator-role.yaml:83]: I thought about it, but having the `apiGroup: ""` not in any of the knative objects, then maybe this role could be set in the camel-k-operator, but having it as you suggest would be fine too.

Comment (astefanutti) [reply config/rbac/operator-role.yaml:83]: I see what you mean. Then having a new role with an explicit name, so it's clear it's not part of the core / mandatory roles, would work even better IMO.

Comment (claudio4j) [reply config/rbac/operator-role.yaml]: For me it's fine having the permission defined in operator-knative.yaml, in the end it's related to the knative interoperability, then it's fine to be there.

**Comment by claudio4j** [comment]
Can we merge it ?

File: config/rbac/operator-role.yaml (hunk 3 above)
Comment (astefanutti) [reply config/rbac/operator-role.yaml]: Sounds good to me too.

**Review by astefanutti — APPROVED** [review]

**Pull request merged**